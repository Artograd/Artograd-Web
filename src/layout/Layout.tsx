import React, { ReactNode } from 'react';
import { Header } from './Header'; // Assuming you create a Header component
import { Footer } from './Footer'; // Assuming you create a Footer component

type LayoutProps = {
  children: ReactNode;
};

export const Layout = ({ children }: LayoutProps) => {
  return (
    <div
      style={{ display: 'flex', flexDirection: 'column', minHeight: '100vh' }}
    >
      <Header />
      <main style={{ flexGrow: 1 }}>{children}</main>
      <Footer />
    </div>
  );
};
