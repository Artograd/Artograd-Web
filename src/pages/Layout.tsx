import React, { ReactNode } from 'react';
import Header from './Header'; // Assuming you create a Header component
import Footer from './Footer'; // Assuming you create a Footer component

type LayoutProps = {
  children: ReactNode;
};

const Layout: React.FC<LayoutProps> = ({ children }) => {
  return (
    <div
      style={{ display: 'flex', flexDirection: 'column', minHeight: '100vh' }}
    >
      <Header />
      <main style={{ flexGrow: 1 }}>{children}</main>
      <Footer />
    </div>
  );
};

export default Layout;
