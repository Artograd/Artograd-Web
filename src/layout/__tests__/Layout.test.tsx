import { render } from '@testing-library/react';
import { Layout } from '../Layout';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: 'localhost:3000/test',
  }),
}));

describe('Layout component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  test('renders correctly', () => {
    const component = render(<Layout>test component</Layout>);

    expect(component).toMatchSnapshot();
  });
});
