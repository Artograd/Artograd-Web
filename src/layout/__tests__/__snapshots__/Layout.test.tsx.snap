// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="display: flex; flex-direction: column; min-height: 100vh;"
      >
        <div
          class=""
          style="min-width: 0; flex-basis: 0px;"
        >
          <nav
            class="cIyvAa uui-main_menu uui-mainmenu-container Qonjm4"
          >
            <div
              class="QFvWHT rnyczA"
            >
              <div
                class="uui-flex-row TtA74w _3odmFf"
              >
                <div
                  data-testid="header-logo-image"
                >
                  <a
                    aria-disabled="true"
                    class="uui-button-box uui-disabled -clickable uui-anchor grqyCP GiETOh"
                    tabindex="-1"
                  >
                    <img
                      alt="Main Menu Logo"
                      class="KnIMMk"
                      src="artograd.logo.svg"
                    />
                  </a>
                </div>
                <button
                  class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- menuPageLink"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="uui-caption"
                  >
                    Home
                  </div>
                </button>
                <button
                  class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- menuPageLink"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="uui-caption"
                  >
                    Tenders
                  </div>
                </button>
                <button
                  class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- menuPageLink"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="uui-caption"
                  >
                    Proposals
                  </div>
                </button>
                <div
                  class="G9zrcQ"
                />
                <div
                  class="uui-flex-row wuDha0 ICVEDz XZX1bt guPk7v YsGDB3 TtA74w _3odmFf"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- languageSelector"
                    role="menuitem"
                    tabindex="0"
                    type="button"
                  >
                    <div
                      class="uui-caption"
                    >
                      en
                    </div>
                    <div
                      class="kb-S-l uui-icon uui-enabled"
                    >
                      <svg
                        class=""
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </div>
                  </button>
                </div>
                <div
                  class="uui-flex-row wuDha0 ICVEDz XZX1bt guPk7v YsGDB3 TtA74w _3odmFf"
                >
                  <a
                    class="uui-button-box uui-enabled -clickable uui-anchor EBheCm pSu6kE uui-button uui-fill-none uui-color-primary uui-size-36 signInButton"
                    href="https://artograd-dev.auth.eu-central-1.amazoncognito.com/login?client_id=3ejpv53oulc3ba35cnrr031ff2&response_type=code&scope=email+openid+phone+profile&redirect_uri=http%3A%2F%2Flocalhost%2Fcallback"
                    tabindex="0"
                  >
                    <div
                      class="uui-caption"
                    >
                      Sign In
                    </div>
                  </a>
                </div>
                <div
                  class="uui-flex-row wuDha0 ICVEDz XZX1bt guPk7v YsGDB3 TtA74w _3odmFf"
                >
                  <a
                    class="uui-button-box uui-enabled -clickable uui-anchor EBheCm pSu6kE uui-button uui-fill-solid uui-color-primary uui-size-36 signUpButton"
                    href="https://artograd-dev.auth.eu-central-1.amazoncognito.com/signup?client_id=3ejpv53oulc3ba35cnrr031ff2&response_type=code&scope=email+openid+phone+profile&redirect_uri=http%3A%2F%2Flocalhost%2Fcallback"
                    tabindex="0"
                  >
                    <div
                      class="uui-caption"
                    >
                      Sign Up
                    </div>
                  </a>
                </div>
              </div>
            </div>
          </nav>
        </div>
        <main
          style="flex-grow: 1;"
        >
          test component
        </main>
        <div
          class=""
          style="min-width: 0; flex-basis: 0px;"
        >
          <nav
            class="cIyvAa uui-main_menu uui-mainmenu-container Qonjm4"
          >
            <div
              class="QFvWHT rnyczA"
            >
              <div
                class="uui-flex-row TtA74w _3odmFf"
              >
                <div
                  data-testid="footer-logo-image"
                >
                  <a
                    aria-disabled="true"
                    class="uui-button-box uui-disabled -clickable uui-anchor grqyCP GiETOh"
                    tabindex="-1"
                  >
                    <img
                      alt="Main Menu Logo"
                      class="KnIMMk"
                      src="artograd.logo.svg"
                    />
                  </a>
                </div>
                <div
                  class="G9zrcQ"
                />
                <span
                  class="uui-button-box uui-enabled _9OkE-9 d7THM-"
                  role="menuitem"
                  tabindex="-1"
                >
                  <div
                    class="uui-caption"
                  >
                    © 1993—2024 EPAM Systems. All Rights Reserved.
                  </div>
                </span>
                <div
                  class="G9zrcQ"
                />
                <button
                  class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- footerLink"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="uui-caption"
                  >
                    Privacy policy
                  </div>
                </button>
                <button
                  class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- footerLink"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="uui-caption"
                  >
                    Cookies policy
                  </div>
                </button>
              </div>
            </div>
          </nav>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="display: flex; flex-direction: column; min-height: 100vh;"
    >
      <div
        class=""
        style="min-width: 0; flex-basis: 0px;"
      >
        <nav
          class="cIyvAa uui-main_menu uui-mainmenu-container Qonjm4"
        >
          <div
            class="QFvWHT rnyczA"
          >
            <div
              class="uui-flex-row TtA74w _3odmFf"
            >
              <div
                data-testid="header-logo-image"
              >
                <a
                  aria-disabled="true"
                  class="uui-button-box uui-disabled -clickable uui-anchor grqyCP GiETOh"
                  tabindex="-1"
                >
                  <img
                    alt="Main Menu Logo"
                    class="KnIMMk"
                    src="artograd.logo.svg"
                  />
                </a>
              </div>
              <button
                class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- menuPageLink"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  Home
                </div>
              </button>
              <button
                class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- menuPageLink"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  Tenders
                </div>
              </button>
              <button
                class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- menuPageLink"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  Proposals
                </div>
              </button>
              <div
                class="G9zrcQ"
              />
              <div
                class="uui-flex-row wuDha0 ICVEDz XZX1bt guPk7v YsGDB3 TtA74w _3odmFf"
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="true"
                  class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- languageSelector"
                  role="menuitem"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="uui-caption"
                  >
                    en
                  </div>
                  <div
                    class="kb-S-l uui-icon uui-enabled"
                  >
                    <svg
                      class=""
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </div>
                </button>
              </div>
              <div
                class="uui-flex-row wuDha0 ICVEDz XZX1bt guPk7v YsGDB3 TtA74w _3odmFf"
              >
                <a
                  class="uui-button-box uui-enabled -clickable uui-anchor EBheCm pSu6kE uui-button uui-fill-none uui-color-primary uui-size-36 signInButton"
                  href="https://artograd-dev.auth.eu-central-1.amazoncognito.com/login?client_id=3ejpv53oulc3ba35cnrr031ff2&response_type=code&scope=email+openid+phone+profile&redirect_uri=http%3A%2F%2Flocalhost%2Fcallback"
                  tabindex="0"
                >
                  <div
                    class="uui-caption"
                  >
                    Sign In
                  </div>
                </a>
              </div>
              <div
                class="uui-flex-row wuDha0 ICVEDz XZX1bt guPk7v YsGDB3 TtA74w _3odmFf"
              >
                <a
                  class="uui-button-box uui-enabled -clickable uui-anchor EBheCm pSu6kE uui-button uui-fill-solid uui-color-primary uui-size-36 signUpButton"
                  href="https://artograd-dev.auth.eu-central-1.amazoncognito.com/signup?client_id=3ejpv53oulc3ba35cnrr031ff2&response_type=code&scope=email+openid+phone+profile&redirect_uri=http%3A%2F%2Flocalhost%2Fcallback"
                  tabindex="0"
                >
                  <div
                    class="uui-caption"
                  >
                    Sign Up
                  </div>
                </a>
              </div>
            </div>
          </div>
        </nav>
      </div>
      <main
        style="flex-grow: 1;"
      >
        test component
      </main>
      <div
        class=""
        style="min-width: 0; flex-basis: 0px;"
      >
        <nav
          class="cIyvAa uui-main_menu uui-mainmenu-container Qonjm4"
        >
          <div
            class="QFvWHT rnyczA"
          >
            <div
              class="uui-flex-row TtA74w _3odmFf"
            >
              <div
                data-testid="footer-logo-image"
              >
                <a
                  aria-disabled="true"
                  class="uui-button-box uui-disabled -clickable uui-anchor grqyCP GiETOh"
                  tabindex="-1"
                >
                  <img
                    alt="Main Menu Logo"
                    class="KnIMMk"
                    src="artograd.logo.svg"
                  />
                </a>
              </div>
              <div
                class="G9zrcQ"
              />
              <span
                class="uui-button-box uui-enabled _9OkE-9 d7THM-"
                role="menuitem"
                tabindex="-1"
              >
                <div
                  class="uui-caption"
                >
                  © 1993—2024 EPAM Systems. All Rights Reserved.
                </div>
              </span>
              <div
                class="G9zrcQ"
              />
              <button
                class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- footerLink"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  Privacy policy
                </div>
              </button>
              <button
                class="uui-button-box uui-enabled -clickable _9OkE-9 d7THM- footerLink"
                role="menuitem"
                tabindex="0"
                type="button"
              >
                <div
                  class="uui-caption"
                >
                  Cookies policy
                </div>
              </button>
            </div>
          </div>
        </nav>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByAria": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByRoleAndText": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByAria": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByRoleAndText": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByAria": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByRoleAndText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByAria": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByRoleAndText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByAria": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByRoleAndText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByAria": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByRoleAndText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
